### PR TITLE
Update deprecated .toIterator with .iterator

### PR DESCRIPTION
**Pull request by nevillelyh**

### CODE REVIEW

File: scio-extra/src/main/scala-2.12/kantan/codecs/compat.scala [hunk 1]
```diff
@@ -6,7 +6,7 @@ object compat {
 
   implicit class ResourceIteratorCompatOps[T](private val underlying: ResourceIterator[T])
       extends AnyVal {
-    def iterator: Iterator[T] = underlying.toIterator
+    def iterator: Iterator[T] = underlying.iterator
```
Comment (regadas) [inline scio-extra/src/main/scala-2.12/kantan/codecs/compat.scala:9]: this one needs to be `toIterator`

File: scio-repl/src/main/scala-2.12/com/spotify/scio/repl/compat/CsvReaderOps.scala [hunk 2]
```diff
@@ -4,6 +4,6 @@ import kantan.csv.CsvReader
 
 final class CsvReaderOps[A](val reader: CsvReader[A]) extends AnyVal {
 
-  def iterator: Iterator[A] = reader.toIterator
+  def iterator: Iterator[A] = reader.iterator
```
Comment (regadas) [inline scio-repl/src/main/scala-2.12/com/spotify/scio/repl/compat/CsvReaderOps.scala:7]: same here

Comment (nevillelyh) [reply scio-repl/src/main/scala-2.12/com/spotify/scio/repl/compat/CsvReaderOps.scala:7]: Seems to work for me locally will double check.

Comment (nevillelyh) [reply scio-repl/src/main/scala-2.12/com/spotify/scio/repl/compat/CsvReaderOps.scala:7]: Yeah this is fine. There was another breakage but all fixed.

Comment (regadas) [reply scio-repl/src/main/scala-2.12/com/spotify/scio/repl/compat/CsvReaderOps.scala:7]: ahhh cool! 😄 sorry about this I confused this with the one provided by `kantan` 🤦

**Pull request merged**